lib/models/os.coffee

```diff
@@ -58,16 +58,10 @@ getOsModel = (deps, opts) ->
 	getDeviceTypes = once ->
 		configModel().getDeviceTypes()
 
-	isValidDeviceType = (deviceType) ->
+	getValidatedDeviceType = (deviceTypeSlug) ->
 		getDeviceTypes()
 		.then (types) ->
-			!!deviceTypesUtil.findBySlug(types, deviceType)
-
-	validateDeviceType = (deviceType) ->
-		isValidDeviceType(deviceType)
-		.then (isValid) ->
-			if not isValid
-				throw new errors.BalenaInvalidDeviceType('No such device type')
```

Comment:
Since I had to do something like this in  the `device.coffee` I grabbed the opportunity and unified it in a single function in the device type helper (which is also in TS).